frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.unit.spec.tsx

```diff
@@ -120,5 +163,104 @@ describe("ActionCreator", () => {
 
       expect(screen.getByText("FooBar")).toBeInTheDocument();
     });
+
+    describe("admin users and has public sharing enabled", () => {
+      const mockUuid = "mock-uuid";
+
+      it("should show action settings button", async () => {
+        await setupEditing({
+          isAdmin: true,
+          isPublicSharingEnabled: true,
+        });
+
+        expect(getIcon("gear", "button")).toBeInTheDocument();
+      });
+
+      it("should be able to enabled action public sharing", async () => {
+        await setupEditing({
+          isAdmin: true,
+          isPublicSharingEnabled: true,
+        });
+
+        getIcon("gear", "button").click();
+
+        expect(screen.getByText("Action settings")).toBeInTheDocument();
+        expect(
+          screen.getByRole("switch", { name: "Make public" }),
+        ).not.toBeChecked();
+        expect(
+          screen.queryByRole("textbox", { name: "Public action link URL" }),
+        ).not.toBeInTheDocument();
+
+        // Make the action public
+        screen.getByRole("switch", { name: "Make public" }).click();
```

Comment:
I think the `name` here makes it clear what's going on; should we remove the comment?